py3status/modules/screenshot.py

```diff
@@ -4,82 +4,179 @@
+      `upload_path` to be configured.
+    * `"lutim"`: upload on a [Lutim](https://framagit.org/luc/lutim) instance, needs
+      `upload_server` to be configured (URL).
+    * `"custom:path_to_script"`: specify a custom script which will be called after the
+      screenshot has been taken and whose first argument will be the path to the
+      screenshot.
+    * `None`: No post processing. Same effect as parameter `push` set to `False`.
+
+The `push` parameter is kept for compatibility reasons.
+
 Configuration parameters:
-    cache_timeout: how often to update in seconds (default 5)
-    file_length: generated file_name length (default 4)
-    push: True/False if yo want to push your screenshot to your server
+    cache_timeout: how often to update in seconds (default 1)
+    empty_text: text to display when there is nothing to show (default "SHOT")
+    file_length: generated random file_name length (default 4)
+    image_format: format of the screenshot (default 'jpg')
+    logfile: path of the logfile, recommended if you use a image hoster, as you will keep
+        the picture URL and the deletion URL (default None)
+    open_webbrowser: when uploading to an image hoster, open the sent capture in the
+        webbrowser (default True)
+    post_processing: specify what to do with the screenshot (default 'scp')
+    push: if you want to push your screenshot to your server (or whatever your optionnal
+        custom script would do) (default True)
+    random_name: generate a random filename for captures, else use date and time
         (default True)
-    save_path: Directory where to store your screenshots. (default '~/Pictures/')
+    save_path: directory where to store your screenshots (default '~/Pictures/')
     screenshot_command: the command used to generate the screenshot
         (default 'gnome-screenshot -f')
     upload_path: the remote path where to push the screenshot (default '/files')
     upload_server: your server address (default 'puzzledge.org')
     upload_user: your ssh user (default 'erol')
 
 Color options:
-    color_good: Displayed color
+    color_done: Displayed color when full operation is over (default color_good)
+    color_processing: color of the text during operation (default color_degraded)
+
+Requires:
+    requests: python module from pypi https://pypi.python.org/pypi/requests only required
+        if you want to use a web API
 
-@author Amaury Brisou <py3status AT puzzledge.org>
+@author Amaury Brisou <py3status AT puzzledge.org>, raspbeguy
 """
 import os
 import random
 import string
 import subprocess
+import webbrowser
+import datetime
 
 
 class Py3status:
     """
     """
     # available configuration parameters
-    cache_timeout = 5
+    cache_timeout = 1
+    empty_text = "SHOT"
     file_length = 4
+    image_format = 'jpg'
+    logfile = None
+    open_webbrowser = True
+    post_processing = 'scp'
     push = True
+    random_name = True
```

Comment:
>  It would be unacceptable to name an image file without its extension.

I'm not sure that I agree.  File extensions is a bit of a windows thing and is not needed, even if it is common.

gnome-screenshot seems to use them but I think that is actually the wrong thing to do and we allow other programs.